### PR TITLE
R1 ICD Update

### PR DESCRIPTION
**Pull request by amsmith-pro**

### CODE REVIEW

File: docs/icd.md [hunk 1]
```diff
@@ -30,35 +30,33 @@ Features of REST:
...
+`server/src/main.rs` | Implementation of REST server interfaces to this microservice
+`client-rest/src/lib.rs` | The `openapi/types.rs` file bundled into a library
 
-Individual services may define further REST endpoints:
+The `client-rest` crate for each microservice (e.g. `svc-cargo-client-rest`) can be published to a registry.
 
 ```tree
 .
-├── Cargo.lock
 ├── Cargo.toml # [workspace] members = ['client', 'server']
```
Comment (GoodluckH) [inline docs/icd.md]: members should be ['client-rest']

File: docs/icd.md [hunk 2]
```diff
@@ -103,58 +101,49 @@ Arrow services, which are written in Rust Programming Language, access gRPC feat
...
+`server/src/grpc.rs` | Autogenerated Server Stubs
+`server/src/main.rs` | Implements Server Stubs
+`client-grpc/src/grpc.rs` | Autogenerated Client Code
 
-Service-specific gRPC files will be defined in each service's repository:
 ```tree
 .
-├── Cargo.lock
-├── Cargo.toml # [workspace] members = ['client', 'server']
+├── Cargo.toml
```
Comment (GoodluckH) [inline docs/icd.md]: \# [workspace] members = ['client-grpc', 'server']

And, how does this come together with the REST things above? Do we expect services like Cargo to have members = ['client-grpc', 'client-rest', 'server']?

**Review by GoodluckH — CHANGES_REQUESTED** [review]

File: docs/icd.md (hunk 1 above)
Comment (amsmith-pro) [reply docs/icd.md]: https://github.com/Arrow-air/se-services/pull/5/commits/c86bba0ad757c2cf45365714b93776bb230e5429

File: docs/icd.md (hunk 2 above)
Comment (amsmith-pro) [reply docs/icd.md]: Updated to show that the REST microservice has a client-grpc as well

All services have grpc interfaces, some services have REST interfaces

https://github.com/Arrow-air/se-services/pull/5/commits/c86bba0ad757c2cf45365714b93776bb230e5429

**Review by GoodluckH — DISMISSED** [review]

**Comment by amsmith-pro** [comment]
Fixed cspell and link errors: https://github.com/Arrow-air/se-services/pull/5/commits/e552e9633b25f0601cd06ea0bbd9ee7b548d6c13

**Review by GoodluckH — APPROVED** [review]

**Review by owlot — APPROVED** [review]

**Pull request merged**